lib/services/watcher.go

```diff
@@ -462,6 +466,15 @@ func (p *proxyCollector) initializationChan() <-chan struct{} {
 	return p.initializationC
 }
 
+func (p *proxyCollector) isInitialized() bool {
+	select {
+	case <-p.initializationC:
+		return true
+	default:
+		return false
+	}
+}
```

Comment:
I'm not familiar with this part of the codebase, but at first look it seems like this might disrupt the logic here
https://github.com/gravitational/teleport/blob/eb48c8ef55b6714df95fe2673a4a25f4cf7eb3de/lib/services/watcher.go#L181

because `initializationC` is only ever written to once

https://github.com/gravitational/teleport/blob/eb48c8ef55b6714df95fe2673a4a25f4cf7eb3de/lib/services/watcher.go#L411-L416

Reply:
TIL that `close` behavior is different from `send`

https://go.dev/play/p/z932MncG1MV